core/store/orm/orm.go

```diff
@@ -326,6 +329,47 @@ func (orm *ORM) LinkEarnedFor(spec *models.JobSpec) (*assets.Link, error) {
 	return earned, nil
 }
 
+// UpsertErrorFor upserts a JobSpecError record, incrementing the occurances counter by 1
+// if the record is found
+func (orm *ORM) UpsertErrorFor(jobID *models.ID, description string) error {
+	orm.MustEnsureAdvisoryLock()
+	err := func() error {
+		foundJobSpecErr, err := orm.FindJobSpecError(jobID, description)
+		if err != nil && err != gorm.ErrRecordNotFound {
+			return err
+		} else if err != nil {
+			newJobSpecError := models.NewJobSpecError(jobID, description)
+			return orm.db.Create(&newJobSpecError).Error
+		} else {
+			foundJobSpecErr.Occurances++
+			return orm.db.Save(&foundJobSpecErr).Error
```

Comment:
This is racy (also, not really an upsert).

Can you use `INSERT INTO ... ON CONFLICT UPDATE` instead?